fdbserver/workloads/ApiWorkload.actor.cpp

```diff
@@ -249,6 +260,18 @@ Key ApiWorkload::generateKey(VectorRef<KeyValueRef> const& data,
 			keyBuffer[0] = deterministicRandom()->randomInt(0, 255);
 	}
 
+	// If encryption validation is enabled; slip "marker pattern" at random location in generated key
+	if (g_simulator->dataAtRestPlaintextMarker.present() &&
+	    keyLength + 1 > g_simulator->dataAtRestPlaintextMarker.get().size()) {
+		int len = keyLength - g_simulator->dataAtRestPlaintextMarker.get().size();
+		// Avoid updating the first byte of the key
+		int idx = len > 1 ? deterministicRandom()->randomInt(1, len) : 1;
```

Comment:
If tenants are enabled then isn't this an issue if we are overriding the first 8 bytes of the key?

Reply:
Maybe we should only modify the value to inject the plaintext marker?

Reply:
You have to check `g_network->isSimulated()` here too at the start of the `if`.

Reply:
Is this transaction using raw mode?  My understanding was if you declare a Tenant for a transaction, all your KV pairs are automatically under that Tenant, you don't / can't write the prefix yourself, that would only be for raw mode.

Reply:
I don't see "tenant" in any code here so I guess that means raw mode?

Reply:
Yeah actually i guess if this is raw access it will automatically append the 8 byte prefix for the default tenant. I would double check to make sure its in raw access mode

Reply:
> If tenants are enabled then isn't this an issue if we are overriding the first 8 bytes of the key?

If tenants are not `enabled` then we don't have `encryption support` as well, hence, the code is not effective. In other words, the code is enabled _iff_ encryption is enabled, hence, `dataAtRestPlaintextMarker` is set. 
Also, keys generated are random and for `TenantMode::required` there is already code that uses `default domain` already created by the simulation test (add tenantPrefix), hence, the code here is basically just inserting marker.